src/components/data-pane/wildcard-field-dropper.tsx

```diff
@@ -0,0 +1,100 @@
+import {Schema} from 'compassql/build/src/schema';
+import {isWildcard} from 'compassql/build/src/wildcard';
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import {ConnectDropTarget, DropTarget, DropTargetCollector, DropTargetSpec} from 'react-dnd';
+import {CUSTOM_WILDCARD_ADD_FIELD, CUSTOM_WILDCARD_REMOVE,
+        CustomWildcardAction} from '../../actions/custom-wildcard-field';
+import {DraggableType} from '../../constants';
+import {CustomWildcardFieldDef} from '../../models/custom-wildcard-field';
+import {DraggedFieldIdentifier} from '../field/index';
+import * as styles from './wildcard-field-dropper.scss';
+
+
+export interface CustomWildcardFieldDropTargetProps {
+  connectDropTarget: ConnectDropTarget;
+  isOver: boolean;
+  item: Object;
+  canDrop: boolean;
+}
+
+export interface CustomWildcardFieldPropsBase {
+  customWildcardField: CustomWildcardFieldDef;
+  schema: Schema;
+  index: number;
+  handleAction?: (action: CustomWildcardAction) => void;
+}
+
+interface CustomWildcardFieldProps extends CustomWildcardFieldDropTargetProps, CustomWildcardFieldPropsBase {};
+
+class CustomWildcardFieldBase extends React.PureComponent<CustomWildcardFieldProps, {}> {
+  public constructor(props: CustomWildcardFieldProps) {
+    super(props);
+    this.customWildcardRemove = this.customWildcardRemove.bind(this);
+  }
+
+  public render() {
+    const {connectDropTarget} = this.props;
+
+    return connectDropTarget(this.props.children[0]);
+  }
+
+  private customWildcardRemove() {
+    const {handleAction, index} = this.props;
+    handleAction({
+      type: CUSTOM_WILDCARD_REMOVE,
+      payload: {
+        index
+      }
+    });
+  }
+}
+
+const customWildcardFieldTarget: DropTargetSpec<CustomWildcardFieldProps> = {
+  drop(props, monitor) {
+    if (monitor.didDrop()) {
+      return;
+    }
+    const {fieldDef} = monitor.getItem() as DraggedFieldIdentifier;
+    const {handleAction, schema, customWildcardField, index} = props;
+
+    const type = customWildcardField.type;
+    if (type === fieldDef.type) {
+      let fields: string[];
+      if (isWildcard(fieldDef.field)) {
+        if (fieldDef.field === '?') {
+          fields = schema.fieldNames()
+                          .filter(field => schema.vlType(field) === type);
```

Comment:
I would just tab by two empty space.  (The current tabbing isn't aligned anyway)